a11ym

```diff
@@ -190,19 +188,34 @@ urls.drain = function () {
     }
 };
 
-var crawler             = new Crawler(url.hostname);
-crawler.initialPath     = url.path;
-crawler.initialPort     = url.port || 80;
-crawler.initialProtocol = (url.protocol || 'http').replace(':', '');
-crawler.interval        = 100;
-crawler.maxConcurrency  = 5;
-crawler.maxDepth        = +program.maximumDepth;
+var parseURL = function (url) {
+    var parsed = URL.parse(url);
+
+    if (!parsed.protocol) {
+        parsed.protocol = 'http';
+    } else {
+        parsed.protocol = parsed.protocol.replace(':', '');
+    }
+
+    if (!parsed.port) {
+        parsed.port = 'http' === parsed.protocol ? 80 : 443;
+    }
+
+    return parsed;
+};
+
+var crawler            = new Crawler();
+crawler.interval       = 100;
+crawler.maxConcurrency = 5;
+crawler.maxDepth       = +program.maximumDepth;
+crawler.filterByDomain = true;
+
+var maximumUrls = +program.maximumUrls;
```

Comment:
yup, right, didn't understand this line correctly... Maybe just increment the number if we reached the "limit", otherwise the value would not really reflect anything that makes sense, but I am being nit-picky